Content.Shared/Ninja/Systems/SharedNinjaSystem.cs

```diff
@@ -0,0 +1,185 @@
+using Content.Shared.Examine;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Popups;
+using Content.Shared.Stealth;
+using Content.Shared.Stealth.Components;
+using Robust.Shared.Network;
+
+namespace Content.Shared.Ninja.Systems;
+
+public abstract class SharedNinjaSystem : EntitySystem
+{
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] protected readonly SharedPopupSystem _popups = default!;
+    [Dependency] protected readonly SharedStealthSystem _stealth = default!;
+
+    public override void Initialize()
+    {
+    	base.Initialize();
+
+        SubscribeLocalEvent<SpaceNinjaGlovesComponent, GotEquippedEvent>(OnGlovesEquipped);
+        SubscribeLocalEvent<SpaceNinjaGlovesComponent, ExaminedEvent>(OnGlovesExamine);
+        SubscribeLocalEvent<SpaceNinjaGlovesComponent, ToggleNinjaGlovesEvent>(OnToggleGloves);
+        SubscribeLocalEvent<SpaceNinjaGlovesComponent, GotUnequippedEvent>(OnGlovesUnequipped);
+
+        SubscribeLocalEvent<SpaceNinjaSuitComponent, GotEquippedEvent>(OnSuitEquipped);
+        SubscribeLocalEvent<SpaceNinjaSuitComponent, GotUnequippedEvent>(OnSuitUnequipped);
+    }
+
+    private void OnGlovesEquipped(EntityUid uid, SpaceNinjaGlovesComponent comp, GotEquippedEvent args)
+    {
+        var user = args.Equipee;
+        if (TryComp<SpaceNinjaComponent>(user, out var ninja))
+	        NinjaEquippedGloves(uid, comp, user, ninja);
+	}
+
+    private void OnGlovesExamine(EntityUid uid, SpaceNinjaGlovesComponent comp, ExaminedEvent args)
+    {
+        if (!args.IsInDetailsRange)
+            return;
+
+		var enabled = HasComp<GlovesEnabledComponent>(uid);
+        args.PushText(Loc.GetString(enabled ? "ninja-gloves-examine-on" : "ninja-gloves-examine-off"));
+    }
+
+    private void OnToggleGloves(EntityUid uid, SpaceNinjaGlovesComponent comp, ToggleNinjaGlovesEvent args)
+    {
+        var user = args.Performer;
+        // need to wear suit to enable gloves
+        if (!TryComp<SpaceNinjaComponent>(user, out var ninja)
+            || ninja.Suit == null
+            || !HasComp<SpaceNinjaSuitComponent>(ninja.Suit.Value))
+        {
+            ClientPopup(Loc.GetString("ninja-gloves-not-wearing-suit"), user);
+            return;
+        }
+
+		var enabled = !HasComp<GlovesEnabledComponent>(uid);
+		if (enabled)
+			AddComp<GlovesEnabledComponent>(uid);
+		else
+			RemComp<GlovesEnabledComponent>(uid);
+
```

Comment:
Same here, replace this with a bool on NinjaGlovesComp.

Reply:
above